airflow/models/dag.py

```diff
@@ -585,6 +587,8 @@ def __init__(
         self.timetable: Timetable
         self.schedule_interval: ScheduleInterval
         self.dataset_triggers: DatasetBooleanCondition | None = None
+        if isinstance(schedule, DatasetsExpression):
+            self.schedule = extract_datasets(dataset_expression=schedule)
         if isinstance(schedule, (DatasetAll, DatasetAny)):
             self.dataset_triggers = schedule
```

Comment:
I wonder if we could unify those three into one base class. This doesn’t need to be done now.

Reply:
Sure I can do this as part of separate PR